### PR TITLE
streamers: use channel mask for zero patterns

### PR DESCRIPTION
**Pull request by jorendumoulin**
If we generate a zero pattern with the streamers, we can use the optional channel mask to make sure we do not burden the tcdm with zero requests.

### CODE REVIEW

File: compiler/accelerators/snax.py [hunk 1]
```diff
@@ -112,8 +116,20 @@ def _generate_streamer_setup_vals(
         result: Sequence[tuple[Sequence[Operation], SSAValue]] = []
 
         for operand, streamer in enumerate(self.streamer_config.data.streamers):
+            # streamer must generate zero pattern if it is set to c0
```
Comment (JosseVanDelm) [inline compiler/accelerators/snax.py:119]:
```suggestion
            # streamer must generate zero pattern if the stream is coming from c0
```

File: compiler/transforms/stream_snaxify.py [hunk 2]
```diff
@@ -223,7 +223,8 @@ def generate_one_list(n: int, i: int):
 
                 # point C to zero allocated row in TCDM.
```
Comment (JosseVanDelm) [inline compiler/transforms/stream_snaxify.py:224]: is this comment still valid?

**Review by JosseVanDelm — APPROVED** [review]
Very nice!

File: compiler/accelerators/snax.py [hunk 3]
```diff
@@ -90,6 +92,8 @@ class SNAXStreamer(ABC):
     streamer_setup_fields: Sequence[str]
     streamer_launch_fields: Sequence[str]
 
+    zero_address = 0x1000_0040
+
```
[the remark this answers is not in the document]
Comment (jorendumoulin) [reply compiler/accelerators/snax.py:96]: unfortunately it must be se tot this value (first row of tcdm which is kept free for zeros)

**Pull request merged**